### PR TITLE
Build changes related to Msal.Broker project

### PR DESCRIPTION
**Pull request by pmaytak**
**Changes proposed in this request**
- Added a build step to pack and sign Msal.Broker project.
- Changed the versioning of Msal.Broker to match Msal project with `-preview` (`$(MsalClientSemVer)-preview`).
   - Main reason is that Msal.Broker is always released together with Msal and Msal.Desktop, so having a separate version can be confusing.
- Renamed the folder containing project from 'Microsoft.Identity.Client.Brokers` to `Microsoft.Identity.Client.Broker`.
   - Helps with build paths.

### CODE REVIEW

**Review by bgavrilMS — APPROVED** [review]

File: src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj [hunk 1]
```diff
@@ -14,10 +14,9 @@
 
   <PropertyGroup Label="NuGet and AssemblyInfo metadata">
     <!--This should be passed from the VSTS build-->
-    <BrokerPackageSemVer Condition="'$(BrokerPackageSemVer)' == ''">0.1.0</BrokerPackageSemVer>
-    <BrokerPackageSemVer>$(BrokerPackageSemVer)</BrokerPackageSemVer>
+    <MsalClientSemVer Condition="'$(MsalClientSemVer)' == ''">4.7.1-preview</MsalClientSemVer>
     <!--This will generate AssemblyVersion, AssemblyFileVersion and AssemblyInformationVersion-->
-    <Version>$(BrokerPackageSemVer)</Version>
+    <Version>$(MsalClientSemVer)-preview</Version>
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj:19]: If we ever plan to release just the broker what will this version be?

**Review by gladjohn — APPROVED** [review]
looks good

File: src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj:19]: We'll have to override to somtheing like `-preview2`

Comment (pmaytak) [reply src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj:19]: What is the expected/common scenario?
1. Msal.Broker always released in sync with Msal main? This is what we do for Msal.Desktop, even if there are no changes to main Msal. 
2.  Msal.Broker will be released out-of-sync with Msal main? In this case it's own version might be better. But then what happens when Msal.Broker goes public - does the version suddenly jump from 0.x.0 to 4.x.x?
3. A hybrid of 1 and 2 above?

Comment (pmaytak) [reply src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj:19]: As discussed, will merge as-is.

**Comment by trwalke** [comment]
Can you post a link to a build run that has the new broker package built?

File: LibsAndSamples.sln [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 17
-VisualStudioVersion = 17.1.32210.238
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.32428.217
```
Comment (trwalke) [inline LibsAndSamples.sln:3]: I wonder why this would be a downgrade?
Is this needed?

**Review by trwalke — APPROVED** [review]

File: LibsAndSamples.sln (hunk 2 above)
Comment (pmaytak) [reply LibsAndSamples.sln:3]: I use VS2019. It's autogenerated and don't think this affects anything.

**Comment by pmaytak** [comment]
> Can you post a link to a build run that has the new broker package built?

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3318/checks?check_run_id=6481997252

**Pull request merged**